drivers/st25r3911b/st25r3911b.c

```diff
@@ -0,0 +1,570 @@
+		    POWER_SUPP_MEAS_FRACTION_DIVISOR / 2) /
+		    POWER_SUPP_MEAS_FRACTION_DIVISOR;
+
+	*voltage = result;
+
+	LOG_DBG("Measured supply voltage %u mV", result);
+
+	return 0;
+}
+
+static int calibrate_antenna(void)
+{
+	int err;
+	u8_t reg = 0;
+	u32_t status;
+
+	err = st25r3911b_reg_read(ST25R3911B_REG_ANTENNA_CAL_CTRL, &reg);
+	if (err) {
+		return err;
+	}
+
+	/* Check if is possible to automatic calibration. */
+	if (reg & ST25R3911B_REG_ANTENNA_CAL_CTRL_TRIM_S) {
+		LOG_ERR("Antenna auto calibration is not possible");
+		return -EACCES;
+	}
+
+	err = st25r3911b_irq_enable(ST25R3911B_IRQ_MASK_DCT);
+	if (err) {
+		return err;
+	}
+
+	err = st25r3911b_cmd_execute(ST25R3911B_CMD_CALIBRATE_ANTENNA);
+	if (err) {
+		st25r3911b_irq_disable(ST25R3911B_IRQ_MASK_DCT);
+		return err;
+	}
+
+	LOG_DBG("Wait for antenna calibration %u ms", T_COMMON_CMD);
+
+	status = st25r3911b_irq_wait_for_irq(ST25R3911B_IRQ_MASK_DCT,
+					     T_COMMON_CMD);
+
+	err = st25r3911b_irq_disable(ST25R3911B_IRQ_MASK_DCT);
+	if (err) {
+		return err;
+	}
+
+	if (!status) {
+		return -ENXIO;
+	}
+
+	LOG_DBG("Antenna calibration success");
+
+	return 0;
+}
+
+static int rx_tx_enable(void)
+{
+	u8_t mask = ST25R3911B_REG_OP_CTRL_TX_EN |
+		    ST25R3911B_REG_OP_CTRL_RX_EN;
+
+	return st25r3911b_reg_modify(ST25R3911B_REG_OP_CTRL, 0, mask);
+}
+
+static int adjust_regulator(u32_t *mV_adjust)
+{
+	int err;
+	u8_t reg;
+	u8_t voltage = 0;
+	u8_t conf = 0;
+	u32_t mV;
+	u32_t status;
+
+	err = st25r3911b_reg_read(ST25R3911B_REG_REGULATOR_CTRL, &reg);
+	if (err) {
+		return err;
+	}
+
+	/* Check possibility of auto calibration. This bit should be 0 */
+	if (reg & ST25R3911B_REG_REGULATOR_CTRL_REG_S) {
+		LOG_ERR("Regulator auto calibration is not possible");
+		return -EACCES;
+	}
+
+	err = st25r3911b_irq_enable(ST25R3911B_IRQ_MASK_DCT);
+	if (err) {
+		return err;
+	}
+
+	LOG_DBG("Regulator adjustment");
+
+	err = st25r3911b_cmd_execute(ST25R3911B_CMD_ADJUST_REGULATOR);
+	if (err) {
+		st25r3911b_irq_disable(ST25R3911B_IRQ_MASK_DCT);
+		return err;
+	}
+
+	LOG_DBG("Wait for regulator adjust %u ms", T_COMMON_CMD);
+
+	status = st25r3911b_irq_wait_for_irq(ST25R3911B_IRQ_MASK_DCT,
+					     T_COMMON_CMD);
+
+	err = st25r3911b_irq_disable(ST25R3911B_IRQ_MASK_DCT);
+	if (err) {
+		return err;
+	}
+
+	if (!status) {
+		return -ENXIO;
+	}
+
+	/* Check voltage value in mV. */
+	err = st25r3911b_reg_read(ST25R3911B_REG_REGULATOR_TIM_DISP, &voltage);
+	if (err) {
+		return err;
+	}
+
+	/* Read configuration reg to check power source 3V3 or 5V. */
+	err = st25r3911b_reg_read(ST25R3911B_REG_IO_CONF2, &conf);
+	if (err) {
+		return err;
+	}
+
+	voltage >>= ST25R3911B_REG_REGULATOR_TIM_DISP_REG_POS;
+	voltage -= REGULATOR_INIT_REG_VALUE;
+
+	if (conf & ST25R3911B_REG_IO_CONF2_SUP3V) {
+		mV = POWER_SUPP_3V3_INIT_VAL;
+		mV += voltage * POWER_SUPP_3V3_STEP;
+
+		LOG_DBG("Power supply 3.3V");
+
+	} else {
+		mV = POWER_SUPP_5V_INIT_VAL;
+		mV += POWER_SUPP_5V_STEP * voltage;
```

Comment:
`mV += voltage * POWER_SUPP_5V_STEP;` to match the 3.3V case

btw, maybe the following:
```
static u32_t convert_voltage(u8_t voltage, bool is_5V)
{
   u32_t init_val = (!is_5V) ? POWER_SUPP_3V3_INIT_VAL : POWER_SUPP_5V_INIT_VAL;
   u32_t step_val = (!is_5V) ? POWER_SUPP_3V3_STEP : POWER_SUPP_5V_STEP;

   return init_val + voltage * step_val;
}
```

Reply:
Ok